core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -109,8 +109,9 @@ public static StructLikeSet toEqualitySet(CloseableIterable<StructLike> eqDelete
 
   public static <T> CloseableIterable<T> streamingFilter(CloseableIterable<T> rows,
                                                          Function<T, Long> rowToPosition,
-                                                         CloseableIterable<Long> posDeletes) {
-    return new PositionStreamDeleteFilter<>(rows, rowToPosition, posDeletes);
+                                                         CloseableIterable<Long> posDeletes,
+                                                         boolean keepDeleteRows) {
```

Comment:
I would rather not change this method. Instead, let's introduce a new method to select deleted rows instead.